integration/helper_test.go

```diff
@@ -543,6 +602,146 @@ func setupRestoreFiles(compressionType string, withPlugin bool) {
 	_, _ = f.WriteString(expectedTOC)
 }
 
+func createDataFile(dataFile string, dataLength int) {
+	// Write data file
+	f, err := os.Create(dataFile + ".gz")
+	if err != nil {
+		Fail(fmt.Sprintf("%v", err))
+	}
+	gzipf := gzip.NewWriter(f)
+	// Named pipes can buffer, so we need to write more than the buffer size to trigger flush error
+	customData := "here is some data\n"
+
+	customData += strings.Repeat("a", dataLength)
+	customData += "here is some data\n"
+
+	if _, err := gzipf.Write([]byte(customData)); err != nil {
+		Fail(fmt.Sprintf("%v", err))
+	}
+	gzipf.Close()
+	f.Close()
```

Comment:
why not defer?

Reply:
Putting close in the natural control flow making the core more clean and readable.

Reply:
After calling `Fail` `f` and/or `gzipf` are not closed?